logging/bin/security_create_rbac.sh

```diff
@@ -43,50 +45,71 @@ tenant=${2}
 namespace=$(echo "$namespace"| tr '[:upper:]' '[:lower:]')
 tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
 
-
-create_ktenant_roles=${CREATE_KTENANT_ROLE:-true}
-
-
-if [ -z "$namespace" ]; then
-  log_error "Required argument NAMESPACE no specified"
-  echo  ""
-  show_usage
-  exit 1
-elif [[ "$namespace" =~ -H|--HELP|-h|--help ]]; then
- show_usage
- exit
+if [ "$namespace" == "_all_" ] && [ "$tenant" == "_all_" ]; then
+   log_debug "All cluster access RBACs requested"
+   cluster="true"
+else
+   cluster="false"
+   log_debug "namespace: $namespace tenant: $tenant"
 fi
 
-validateNamespace "$namespace"
+create_ktenant_roles=${CREATE_KTENANT_ROLE:-true}
 
-if [[ "$tenant" =~ -H|--HELP|-h|--help ]]; then
-   show_usage
-   exit
-elif [ -n "$tenant" ]; then
+if [ "$cluster" == "true" ]; then
 
-   validateTenantID $tenant
+   # creating cluster-wide RBACs
+   INDEX_PREFIX=viya_logs
+   INDEX_NST="*"
+   ROLENAME=search_index_-ALL-
+   BE_ROLENAME=V4MCLUSTER_ADMIN_kibana_users
+   NST="cluster_admins"
+   index_role_template="index_role_allcluster.json"
+   kibana_tenant_role_template="kibana_tenant_clusteradmins_role.json"
 
-   NST="${namespace}_${tenant}"
-   INDEX_NST="${namespace}-__${tenant}__"
 else
-   NST="$namespace"
-   INDEX_NST="${namespace}"
-fi
+   # creating namespace or tenant limited RBACs
+   if [ -z "$namespace" ]; then
+     log_error "Required argument NAMESPACE no specified"
+     echo  ""
+     show_usage
+     exit 1
+   elif [[ "$namespace" =~ -H|--HELP|-h|--help ]]; then
+    show_usage
+    exit
+   fi
+
+   validateNamespace "$namespace"
+
+   if [[ "$tenant" =~ -H|--HELP|-h|--help ]]; then
+      show_usage
+      exit
+   elif [ -n "$tenant" ]; then
+
+      validateTenantID $tenant
+
+      NST="${namespace}_${tenant}"
+      INDEX_NST="${namespace}-__${tenant}__"
+   else
+      NST="$namespace"
+      INDEX_NST="${namespace}"
+   fi
+
+   if [ -n "$tenant" ]; then
+     log_notice "Creating access controls for tenant [$tenant] within namespace [$namespace] [$(date)]"
+   else
+     log_notice "Creating access controls for namespace [$namespace] [$(date)]"
```

Comment:
Why is the date important here?

Reply:
In all (most?) of the component deployment scripts on the logging side, I've included the date/time in the first and last messages.  This allows the user to correlate running the script with changes in behavior later. And, if they have logs from several runs saved somewhere, helps clarify which log corresponds to which run.